File: Algos/linear_search.py

data = [10, 5, 2, 0, 9, 6, 8, 1, 14, 3, 4, 7]
target = 11

# linear search
def linear_search(data, target):
    """Linear search algorithm to look for `target` in `data`.

    Args:
        data (list): linear data structure
        target (int): target element to search for
    """
    
    for i in range(len(data)):
        if data[i] == target:
            print(f"Found {target} at index {i}")
            return True
    print(f"Item {target} not found")
    return False
    
linear_search(data=data, target=target)

File: Algos/LinearSearch.java
package Algos;
/**
 * LinearSearch
 */
public class LinearSearch {

    // public LinearSearch() {}

    public boolean search(int[] data, int target) {
        for (int i = 0; i < data.length; i++) {
            if (data[i] == target) {
                System.out.println("Found " + target + " at index " + i);
                return true;
            }
        }
        System.out.println("Item " + target + " not found");
        return false;
    }
}